components/sync15/src/state.rs

```diff
@@ -202,6 +202,24 @@ fn new_global(pgs: &PersistedGlobalState) -> error::Result<MetaGlobalRecord> {
     })
 }
 
+fn fixup_meta_global(global: &mut MetaGlobalRecord) -> bool {
+    let mut changed_any = false;
+    for &(name, version) in DEFAULT_ENGINES.iter() {
```

Comment:
This is where it would be helpful to have `accepted`—we could then repopulate m/g from that. As it is, I think this has the side effect of enabling all default engines, even ones that other clients don't support yet.

Right now, it's more of a theoretical concern, but I'm thinking of what happens when we enable, say, credit cards everywhere. IIUC, this will opt the user in to credit card sync, when they never actually confirmed their selection. But, if we have `accepted` at that point, I _think_ it'll be fine. (That said, how are we going to populate `accepted` without asking every user to confirm their data choices again? Would using m/g bring up this problem?)

We could also ask the Sync Manager to give us a list of supported engines, but `sync15` doesn't know anything about them. Maybe if we passed a slice of engine names in—or used the collection names from the `&[&dyn Store]` that we pass to `synchronize`?

Or am I overthinking this, and using `DEFAULT_ENGINES` is just fine?

Reply:
> But we should, on the very first sync—it's the list of choices we get back from the CWTS flow.

Right, but we're not guaranteed to be here when coming off the CWTS flow.

It's also worth noting that AFAICT an engine being in meta/global doesn't mean it's accepted, it just means we know it exists -- it being in info/collections means it's accepted. I've verified that declining engines doesn't remove anything from meta/global.